node/src/actors/chain_manager/mining.rs

```diff
@@ -406,16 +392,13 @@ impl ChainManager {
                         })
                         .then(|result| match result {
                             // If the result of `RunTally` is `Ok`, it will be published as tally
-                            Ok(Ok(value)) => futures::future::ok(value),
+                            Ok(Some(value)) => futures::future::ok(value),
                             // If the result of `RunTally` is `Err`, we ignore this data request.
                             // If a data request has an invalid tally script, it will never resolve.
                             // The Radon engine should return `Ok(RadonError)` for errors which should
                             // be published as the result of the data request, or we could use a
                             // special radon value to indicate "generic error"
-                            Ok(Err(e)) => {
-                                log::warn!("Couldn't run tally: {}", e);
-                                futures::future::err(())
-                            }
+                            Ok(None) => unreachable!("Couldn't run tally"),
```

Comment:
If None is unreachable, why use an Option?

Reply:
Actix doesn't like handlers returning `RadonReport` :man_shrugging: